src/nvim/api/buffer.c

```diff
@@ -1137,12 +1137,16 @@ Boolean nvim_buf_is_loaded(Buffer buffer)
   return buf && buf->b_ml.ml_mfp != NULL;
 }
 
-/// Deletes the buffer. See |:bwipeout|
+/// Deletes the buffer.
 ///
 /// @param buffer Buffer handle, or 0 for current buffer
 /// @param opts  Optional parameters. Keys:
 ///          - force:  Force deletion and ignore unsaved changes.
-///          - unload: Unloaded only, do not delete. See |:bunload|
+///          - action: (default: "wipeout")
```

Comment:
nit: it's not really an action , i will try to suggest a different name

Reply:
I was going off based on the name of the enum https://github.com/neovim/neovim/blob/081eb72a80afd83398465382171765bdeaf6dba5/src/nvim/buffer.h#L39-L46

Reply:
It's definitely good to follow existing precedent, but for interfaces we often want to use precedent at the interface level (user-facing in this case) rather than internal C idioms.

But I can't think of a much better name. In `:help api` it looks like `type` is fairly common. So this could be named `type`.

Reply:
I feel like `type` in this case is likely to imply the buffer type, rather than the operation type, as if it's a filtering parameter, e.g.

```lua
vim.api.nvim_bufdelete(buffer=0, {type="prompt"})
```